From: Kylie McClain <somasis@exherbo.org>
Date: Tue, 8 Mar 2016 21:40:33 -0500
Subject: syscfg: Add lock-obj-pub files for {armv5, armv6,
 x86_64}-musl targets

* src/syscfg/lock-obj-pub.armv5-unknown-linux-musleabi.h: New.
* src/syscfg/lock-obj-pub.armv6-unknown-linux-musleabihf.h New.
* src/syscfg/lock-obj-pub.x86_64-pc-linux-musl.h: New.
* src/Makefile.am (lock_obj_pub): Add files.

--
This patch adds three new precompiled lock-obj-pub files:
    - armv5-unknown-linux-musleabi
    - armv6-unknown-linux-musleabihf
    - x86_64-pc-linux-musl

ChangeLog lines and Makefile patch by -wk
---
 src/Makefile.am                                    |  3 +++
 .../lock-obj-pub.armv5-unknown-linux-musleabi.h    | 23 ++++++++++++++++++++
 .../lock-obj-pub.armv6-unknown-linux-musleabihf.h  | 23 ++++++++++++++++++++
 src/syscfg/lock-obj-pub.x86_64-pc-linux-musl.h     | 25 ++++++++++++++++++++++
 4 files changed, 74 insertions(+)
 create mode 100644 src/syscfg/lock-obj-pub.armv5-unknown-linux-musleabi.h
 create mode 100644 src/syscfg/lock-obj-pub.armv6-unknown-linux-musleabihf.h
 create mode 100644 src/syscfg/lock-obj-pub.x86_64-pc-linux-musl.h

diff --git a/src/Makefile.am b/src/Makefile.am
index 8aca7df..c1363ae 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -48,6 +48,8 @@ lock_obj_pub = \
         syscfg/lock-obj-pub.arm-unknown-linux-gnueabi.h     \
         syscfg/lock-obj-pub.arm-unknown-linux-gnueabihf.h   \
 	syscfg/lock-obj-pub.arm-apple-darwin.h              \
+        syscfg/lock-obj-pub.armv5-unknown-linux-musleabi.h  \
+        syscfg/lock-obj-pub.armv6-unknown-linux-musleabihf.h \
         syscfg/lock-obj-pub.hppa-unknown-linux-gnu.h        \
         syscfg/lock-obj-pub.i686-pc-gnu.h                   \
         syscfg/lock-obj-pub.i686-pc-kfreebsd-gnu.h          \
@@ -68,6 +70,7 @@ lock_obj_pub = \
         syscfg/lock-obj-pub.x86_64-pc-kfreebsd-gnu.h        \
         syscfg/lock-obj-pub.x86_64-pc-linux-gnu.h           \
         syscfg/lock-obj-pub.x86_64-pc-linux-gnux32.h        \
+        syscfg/lock-obj-pub.x86_64-pc-linux-musl.h          \
 	syscfg/lock-obj-pub.mingw32.h
 
 
diff --git a/src/syscfg/lock-obj-pub.armv5-unknown-linux-musleabi.h b/src/syscfg/lock-obj-pub.armv5-unknown-linux-musleabi.h
new file mode 100644
index 0000000..c7b6165
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.armv5-unknown-linux-musleabi.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.armv5-unknown-linux-musleabi.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.armv6-unknown-linux-musleabihf.h b/src/syscfg/lock-obj-pub.armv6-unknown-linux-musleabihf.h
new file mode 100644
index 0000000..6535a9b
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.armv6-unknown-linux-musleabihf.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.armv6-unknown-linux-musleabihf.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.x86_64-pc-linux-musl.h b/src/syscfg/lock-obj-pub.x86_64-pc-linux-musl.h
new file mode 100644
index 0000000..1b059f4
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.x86_64-pc-linux-musl.h
@@ -0,0 +1,25 @@
+## lock-obj-pub.x86_64-pc-linux-musl.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[40];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
